### PR TITLE
Payments functionality in backend

### PR DESCRIPTION
**Pull request by dr03ramos**
This PR refactors the backend link between rachas and users so now users are able to say they payed the racha and the owner is able to confirm that. Corresponding endpoints were built for those features.

### CODE REVIEW

**Comment by dr03ramos** [comment]
The following is the idea behind payment (in Portuguese):

File: src/main/java/ifsul/agileproject/rachadinha/service/impl/RachaServiceImpl.java [hunk 1]
```diff
@@ -120,16 +159,24 @@ public void addMemberToRacha(Long rachaId, Long userId, String password) {
...
     }
-    racha.getMembers().add(user);
-    user.getRachas().add(racha);
-    rachaRepository.save(racha);
-    userRepository.save(user);
+    Payment payment = Payment.builder()
+        .racha(racha)
+        .user(user)
+        .userSaidPaid(false)
+        .confirmedByOwner(true)
```
Comment (edukatiau) [inline src/main/java/ifsul/agileproject/rachadinha/service/impl/RachaServiceImpl.java:176]: When the user join into a racha, the owner already confirm?

**Review by edukatiau — CHANGES_REQUESTED** [review]
Lets talk

File: src/main/java/ifsul/agileproject/rachadinha/service/impl/RachaServiceImpl.java (hunk 1 above)
Comment (dr03ramos) [reply src/main/java/ifsul/agileproject/rachadinha/service/impl/RachaServiceImpl.java:176]: That is how i did it, but we can change 😅
The owner is still going to be able to say if they have already paid. As `confirmedByOwner` is true by default, they will not have to confirm their own `userSaidPaid` state later.
We can add further logic to this, but i think the current implementation simplifies this process to the owner.

File: src/main/java/ifsul/agileproject/rachadinha/domain/dto/OwnerDTO.java [hunk 2]
```diff
@@ -1,10 +1,19 @@
...
 
-public record OwnerDTO(long id, String name, String email) {
+@Data
+public class OwnerDTO {
 
-  public static OwnerDTO transformaEmDTO(User user){
-    return new OwnerDTO(user.getId(), user.getName(), user.getEmail());
+  private String name;
+  private String email;
+  private String password;
```
[the remark this answers is not in the document]
Comment (dr03ramos) [reply src/main/java/ifsul/agileproject/rachadinha/domain/dto/OwnerDTO.java:12]: You're right and the current implementation is mistaken. I'll correct this in a future commit.

[the remark this answers is not in the document]
Comment (dr03ramos) [reply src/main/java/ifsul/agileproject/rachadinha/domain/dto/OwnerDTO.java]: solved in commit https://github.com/guilhXavier/agile-project-3a/pull/38/commits/7a4ca5b5c760f370c8b75ef3540f4d1673ba9f14

File: src/main/java/ifsul/agileproject/rachadinha/service/impl/RachaServiceImpl.java (hunk 1 above)
Comment (dr03ramos) [reply src/main/java/ifsul/agileproject/rachadinha/service/impl/RachaServiceImpl.java:174]: This behavior should apply only to the owner. Regular users joining the racha should have both `userSaidPaid` and `confirmedByOwner` set as `false`.

**Review by edukatiau — APPROVED** [review]
Lets go boys🚀

**Pull request merged**